### PR TITLE
cython-lint : further fixes in quadratic forms

### PR DESCRIPTION
**Pull request by fchapoton**
This fixes some suggestions of `cython-lint` in `quadratic_forms`.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/quadratic_forms/count_local_2.pyx [hunk 1]
```diff
@@ -209,91 +198,87 @@ cdef local_solution_type_cdef(Q, p, w, zvec, nzvec):
...
-        if (((w[i] % p) != 0)  and ((Q[i,i] % p) != 0)):
+    for i in range(n):
+        if w[i] % p and Q[i, i] % p:
             return <long> 1
-    if (p == 2):
-        for i from 0 <= i < (n - 1):
-            if (((Q[i,i+1] % p) != 0) and (((w[i] % p) != 0) or ((w[i+1] % p) != 0))):
+    if p == 2:
+        for i in range(n - 1):
+            if ((Q[i, i+1] % p)) and (((w[i] % p)) or ((w[i+1] % p))):
```
Comment (mkoeppe) [inline src/sage/quadratic_forms/count_local_2.pyx]: some parens can be peeled off here

**Review by mkoeppe — COMMENTED** [review]
Otherwise LGTM

**Comment by fchapoton** [comment]
done, thanks

File: src/sage/quadratic_forms/count_local_2.pyx [hunk 2]
```diff
@@ -209,91 +198,87 @@ cdef local_solution_type_cdef(Q, p, w, zvec, nzvec):
...
             wS1_nonzero_flag = True
 
     # 4: Check Bad-type I
-    if (wS1_nonzero_flag is True):
+    if wS1_nonzero_flag:
         return <long> 4
 
     # 5: Check Bad-type II
-    if (wS1_nonzero_flag is False):
+    if wS1_nonzero_flag:
```
Comment (mkoeppe) [inline src/sage/quadratic_forms/count_local_2.pyx]: This change looks dubious

Comment (fchapoton) [reply src/sage/quadratic_forms/count_local_2.pyx]: this is never assigned anything but boolean

```
git grep "wS1_non" src/
src/sage/quadratic_forms/count_local_2.pyx:    wS1_nonzero_flag = False
src/sage/quadratic_forms/count_local_2.pyx:            wS1_nonzero_flag = True
src/sage/quadratic_forms/count_local_2.pyx:    if (wS1_nonzero_flag is True):
src/sage/quadratic_forms/count_local_2.pyx:    if (wS1_nonzero_flag is False):
```

Comment (mkoeppe) [reply src/sage/quadratic_forms/count_local_2.pyx]: What I mean is: `wS1_nonzero_flag is False` is equivalent to `not wS1_nonzero_flag`, not `wS1_nonzero_flag`.

Comment (fchapoton) [reply src/sage/quadratic_forms/count_local_2.pyx]: oups, indeed

**Review by mkoeppe — APPROVED** [review]

**Pull request merged**